infrastructure/simulation-tool/test/operations/transfer.test.ts

```diff
@@ -0,0 +1,172 @@
+        transfer.awaitVerifyReceipt.callsFake(() => Promise.resolve(expectedVerifierReceipt));
+
+        const transferResult = await geteTransferResult(transfer);
+
+        expect(transferResult.opL2Receipt).to.eq(expectedL2Receipt);
+        expect(transferResult.verifierReceipt).to.eq(expectedVerifierReceipt);
+    });
+});
+
+describe('generateTransfers', () => {
+    it('should generate expected number of transfers', () => {
+        const numberOfTransfers = 10;
+        const funderL2Wallet = sinon.createStubInstance(RollupWallet);
+        const recepients = [...Array(numberOfTransfers)].map(() => sinon.createStubInstance(EthersWallet));
+        const transfers = generateTransfers(numberOfTransfers, funderL2Wallet, recepients);
+
+        expect(transfers.length).to.eq(numberOfTransfers);
+    });
+
+    it('should generate transfers with funder as sender', () => {
+        const numberOfTransfers = 10;
+        const funderL2Wallet = sinon.createStubInstance(RollupWallet);
+        const recepients = [...Array(numberOfTransfers)].map(() => sinon.createStubInstance(EthersWallet));
+        const transfers = generateTransfers(numberOfTransfers, funderL2Wallet, recepients);
+
+        transfers.forEach((transfer) => {
+            expect(transfer.from).to.eq(funderL2Wallet);
+        });
+    });
+
+    it('should generate transfers with recepients as recipients', () => {
+        const numberOfTransfers = 10;
+        const funderL2Wallet = sinon.createStubInstance(RollupWallet);
+        const recepients = [...Array(numberOfTransfers)].map(() => {
+            const walletStub = sinon.createStubInstance(EthersWallet);
+            (<{ address: string }>walletStub).address = '0x' + Math.random().toString(16).substring(2, 42);
+
+            return walletStub;
+        });
+        const transfers = generateTransfers(numberOfTransfers, funderL2Wallet, recepients);
+        const expectedRecepients = recepients.map((receipient) => receipient.address);
+
+        transfers.forEach((transfer) => {
+            expect(expectedRecepients).to.include(transfer.to);
+        });
+    });
+});
+
+describe('executeTransfers', () => {
+    it('should execute all transfers', async () => {
+        const numberOfTransfers = 10;
+        const funderL2Wallet = sinon.createStubInstance(RollupWallet);
+        funderL2Wallet.syncTransfer.resolves(sinon.createStubInstance(Transaction));
+        const recepients = [...Array(5)].map(() => sinon.createStubInstance(EthersWallet));
+        const transfers = generateTransfers(numberOfTransfers, funderL2Wallet, recepients);
+        const delay = 0;
+        sinon.stub(utils, 'sleep').callsFake(() => Promise.resolve());
+        const executedTransfers = await executeTransfers(transfers, funderL2Wallet, delay);
+
+        expect(executedTransfers.length).to.eq(numberOfTransfers);
+    });
+
+    it('should return executed transfers', async () => {
```

Comment:
Should we check how many txs have been created per second?

Reply:
sure

Reply:
added